### PR TITLE
refactor: system/View/Plugins.php

### PR DESCRIPTION
**Pull request by mcsaygili**
**Description**
See #8732 

- add types

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: writable/debugbar/index.html [hunk 1]
```diff
@@ -1,11 +0,0 @@
...
-<html>
-<head>
-	<title>403 Forbidden</title>
-</head>
-<body>
-
-<p>Directory access is forbidden.</p>
-
-</body>
-</html>
```
Comment (kenjis) [inline writable/debugbar/index.html:11]: Don't remove this file.

File: writable/debugbar/index.html [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
 <head>
-	<title>403 Forbidden</title>
+    <title>403 Forbidden</title>
```
Comment (kenjis) [inline writable/debugbar/index.html:4]: Don't change this file.

**Comment by kenjis** [comment]
The first commit message:
> test: refactor system/View/Plugins.php 

`test:` means the commit changes only test files.
So the message is wrong.

File: system/View/Plugins.php [hunk 3]
```diff
@@ -42,6 +42,8 @@ public static function previousURL()
 
     /**
      * Wrap helper function to use as view plugin.
+     *
+     * @param array{uri?: string, title?: string, attributes?: string} $params
```
Comment (kenjis) [inline system/View/Plugins.php:46]:
```suggestion
     * @param array{email?: string, title?: string, attributes?: array<string, string>|object|string} $params
```

File: system/View/Plugins.php [hunk 4]
```diff
@@ -54,6 +56,8 @@ public static function mailto(array $params = []): string
 
     /**
      * Wrap helper function to use as view plugin.
+     *
+     * @param array{email?: string, title?: string, attributes?: string} $params
```
Comment (kenjis) [inline system/View/Plugins.php:60]:
```suggestion
     * @param array{email?: string, title?: string, attributes?: array<string, string>|object|string} $params
```

File: system/View/Plugins.php [hunk 5]
```diff
@@ -66,6 +70,8 @@ public static function safeMailto(array $params = []): string
 
     /**
      * Wrap helper function to use as view plugin.
+     *
+     * @param list<int|string> $params
```
Comment (kenjis) [inline system/View/Plugins.php:74]:
```suggestion
     * @param array<string, float|int|string>|list<float|int|string> $args
```

Comment (kenjis) [inline system/View/Plugins.php:74]:
```suggestion
     * @param array<int|string, string>|list<string> $params
```

When the template is `{+ lang Number.terabyteAbbr foo=1 +}`,
```php
array (
  0 => 'Number.terabyteAbbr',
  'foo' => '1',
)
```
When `{+ lang Number.terabyteAbbr foo 1 +}`,
```php
array (
  0 => 'Number.terabyteAbbr',
  1 => 'foo',
  2 => '1',
)
```

File: system/View/Plugins.php [hunk 6]
```diff
@@ -99,6 +109,8 @@ public static function route(array $params = [])
 
     /**
      * Wrap helper function to use as view plugin.
+     *
+     * @param array{uri?: string, title?: string, attributes?: string} $params
```
Comment (kenjis) [inline system/View/Plugins.php:113]:
```suggestion
     * @param list<string> $params
```

File: system/View/Plugins.php [hunk 7]
```diff
@@ -90,6 +98,8 @@ public static function ValidationErrors(array $params = []): string
     /**
      * Wrap helper function to use as view plugin.
      *
+     * @param list<int|string> $params
```
Comment (kenjis) [inline system/View/Plugins.php:101]: There seems no way to set int values from templates.
```suggestion
     * @param list<string> $params
```

File: system/View/Plugins.php (hunk 5 above)
Comment (mcsaygili) [reply system/View/Plugins.php:74]: I missed it.

**Review by kenjis — APPROVED** [review]

**Comment by kenjis** [comment]
@mcsaygili Thank you!

**Pull request merged**